src/lib/mongoBackend/collectionOperations.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+*
+* Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Fermín Galán
+*/
+
+#include "mongoBackend/MongoGlobal.h"
+#include "logMsg/traceLevels.h"
+
+#include "mongo/client/dbclient.h"
+
+using namespace mongo;
+
+/* ****************************************************************************
```

Comment:
We have a lot of places in mongoBackend with the same repeated pattern: getConnection / try / catch Exception / catch (...), which make some part of mongoBackend difficult to read. Thus, the idea is to have "wrappers" for query/update/insert/remove/etc and other MongoDB operations in this module, making its usage simpler from the caller side.

I have started to use the pattern in mongoListSubscriptions but at the end all the mongoBackend will be refactored to work in this way. My only doubt if it the name (collectionOperations, as they are operations that can be applied on collections) could be improved. Feedback welcome :)